hooks/tk-multi-publish2/basic/publish_document.py

```diff
@@ -391,4 +392,15 @@ def _document_path(document):
     except Exception:
         path = None
 
+    # The RPC API is always going to return a utf-8 encoded string
+    # here. This is fine (and correct) in almost all cases, but there
+    # are some situations on Windows where the encoded path, if it
+    # includes non-ascii characters, will test as false when checking
+    # existence. In that case, we're forced to decode the string to
+    # unicode.
+    #
+    # Note: This does not appear to be an issue on OS X.
+    if path and sys.platform.startswith("win") and not os.path.exists(path):
+        path = path.decode("utf-8")
```

Comment:
We should add a `:returns:` to the docstring. But if i am reading this correctly, looks like the returns would look like:

```
returns a utf-8 encoded path to the document, UNLESS you are on 
windows and the path does not exist or contains non-ascii characters, 
in which case a unicode string is returned.
```
That doesn't seem like good encapsulation or consistency and seems like it can lead to subtle bugs.

As far as i understand, i think there are two ways to go, both approaches returning a consistent string from `_document_path()`:

- If there are several more complex path operations inside the photoshop hook logic, all requiring unicode paths on windows to function, we return all paths from `_document_path()` as unicode. Ideally for all OSes, but possibly for windows only if that makes sense.

- if path operations are atomic and simple (like for example in the loader), we convert to unicode right before the operation (`os.path.exists(path.decode("utf-8")`)

I am happy either way as long as we keep the return values consistent and we explain the logic in some comments.

Reply:
Makes sense. Thanks!